interop-testing/src/main/java/io/grpc/testing/integration/StressTestClient.java

```diff
@@ -0,0 +1,574 @@
+          }
+        }
+      }
+    } catch (Exception e) {
+      log.log(Level.WARNING, "The stress test client encountered an error!", e);
+    }
+  }
+
+  @VisibleForTesting
+  void blockUntilStressTestComplete() throws Exception {
+    Preconditions.checkState(!shutdown, "client was shutdown.");
+
+    ListenableFuture<?> f = Futures.allAsList(workerFutures);
+    if (durationSecs < 0) {
+      // -1 indicates that the stress test runs "forever".
+      f.get();
+    } else {
+      f.get(durationSecs + WORKER_GRACE_PERIOD_SECS, TimeUnit.SECONDS);
+    }
+  }
+
+  @VisibleForTesting
+  void shutdown() {
+    if (shutdown) {
+      return;
+    }
+    shutdown = true;
+
+    for (ManagedChannel ch : channels) {
+      try {
+        ch.shutdownNow();
+      } catch (Throwable t) {
+        log.log(Level.WARNING, "Error shutting down channel!", t);
+      }
+    }
+
+    try {
+      metricsServer.shutdownNow();
+    } catch (Throwable t) {
+      log.log(Level.WARNING, "Error shutting down metrics service!", t);
+    }
+
+    try {
+      if (threadpool != null) {
+        threadpool.shutdownNow();
+      }
+    } catch (Throwable t) {
+      log.log(Level.WARNING, "Error shutting down threadpool.", t);
+    }
+  }
+
+  private static List<InetSocketAddress> parseServerAddresses(String addressesStr) {
+    List<InetSocketAddress> addresses = new ArrayList<InetSocketAddress>();
+
+    for (String[] namePort : parseCommaSeparatedTuples(addressesStr)) {
+      String name = namePort[0];
+      int port = Integer.valueOf(namePort[1]);
+      addresses.add(new InetSocketAddress(name, port));
+    }
+
+    return addresses;
+  }
+
+  private static List<TestCaseWeightPair> parseTestCases(String testCasesStr) {
+    List<TestCaseWeightPair> testCaseWeightPairs = new ArrayList<TestCaseWeightPair>();
+
+    for (String[] nameWeight : parseCommaSeparatedTuples(testCasesStr)) {
+      TestCases testCase = TestCases.fromString(nameWeight[0]);
+      int weight = Integer.valueOf(nameWeight[1]);
+      testCaseWeightPairs.add(new TestCaseWeightPair(testCase, weight));
+    }
+
+    return testCaseWeightPairs;
+  }
+
+  private static List<String[]> parseCommaSeparatedTuples(String str) {
+    List<String[]> tuples = new ArrayList<String[]>();
+
+    String[] tuples0 = str.split(",");
+    for (String tupleStr : tuples0) {
+      String[] tuple = tupleStr.split(":");
+      if (tuple.length != 2) {
+        throw new IllegalArgumentException("Illegal tuple format: " + tupleStr);
+      }
+      tuples.add(tuple);
+    }
+
+    return tuples;
+  }
+
+  private static ManagedChannel createChannel(InetSocketAddress address) {
+    return NettyChannelBuilder.forAddress(address)
+        .negotiationType(NegotiationType.PLAINTEXT)
+        .build();
+  }
+
+  private static String serverAddressesToString(List<InetSocketAddress> addresses) {
+    return Joiner.on(",").join(Lists.transform(addresses,
+        new Function<InetSocketAddress, String>() {
```

Comment:
Just use a plain for loop and temporary list.  It's shorter and easier to understand.